metric/metricdata/exemplar.go

```diff
@@ -24,10 +31,23 @@ import (
 // Their purpose is to provide an example of the kind of thing
 // (request, RPC, trace span, etc.) that resulted in that measurement.
 type Exemplar struct {
-	Value       float64     // the value that was recorded
-	Timestamp   time.Time   // the time the value was recorded
-	Attachments Attachments // attachments (if any)
+	Value       float64                // the value that was recorded
+	Timestamp   time.Time              // the time the value was recorded
+	Attachments map[string]interface{} // attachments (if any)
+}
+
+// Attachment is a key-value pair associated with a recorded example data point.
+type Attachment struct {
+	Key   string
```

Comment:
`Key` and `Value` have to be exposed because other packages (stats, exporters) need to access them.